src/control/cmd/daos_agent/infocache.go

```diff
@@ -328,7 +322,10 @@ func (c *InfoCache) GetAttachInfo(ctx context.Context, sys string) (*control.Get
 		return nil, errors.Errorf("unexpected attach info data type %T", item)
 	}
 
-	return cai.lastResponse, nil
+	resp := new(control.GetAttachInfoResp)
+	*resp = *cai.lastResponse
```

Comment:
Of note, this is a shallow copy, so if there are concerns about modification of the slice contents, you'll probably want to make a deep copy of those too.

Reply:
The contents I'm worried about modifying are the ClientNetHint, which isn't a pointer. So I think we're okay in this case. Maybe I should add a comment.

Reply:
Actually, I'm considering adding a deep copy function, therefore avoiding any future confusion this might cause. I'll write that and re-push.